### PR TITLE
Create youtube-theater-fix.js, fixes brave/brave-browser#29591

### PR DESCRIPTION
**Pull request by pes10k**
adds a scriptlet to mirror the session cookie value in localstorage

addresses brave/brave-browser#29591

### CODE REVIEW

File: resources/youtube-theater-fix.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        await cookieStore.set(cookieKey, localStorage[storeKey])
+    }
+    setInterval(async _ => {
+        try {
+            const wideCookie = await cookieStore(cookieKey)
+            localStorage[storeKey] = wideCookie.value
+        } catch () {
+            // swallow error from no cookie existing
+        }
+    }, 1000)
```
Comment (ShivanKaul) [inline resources/youtube-theater-fix.js:19]: I think we're missing a closing `)` here.

**Review by antonok-edm — COMMENTED** [review]
you might want to prefix the filename with `brave-` so that only Brave lists will be able to use it

File: resources/youtube-theater-fix.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const storeKey = 'brave::wide'
+    const cookieKey = 'wide'
+    if (localStorage[storeKey] !== undefined) {
+        await cookieStore.set(cookieKey, localStorage[storeKey])
+    }
+    setInterval(async _ => {
+        try {
+            const wideCookie = await cookieStore(cookieKey)
+            localStorage[storeKey] = wideCookie.value
+        } catch () {
```
Comment (ShivanKaul) [inline resources/youtube-theater-fix.js:16]:
```suggestion
        } catch {
```

Comment (ShivanKaul) [reply resources/youtube-theater-fix.js]: (Just added a placeholder arg)

**Comment by ShivanKaul** [comment]
We also need a corresponding filter list rule: https://github.com/brave/adblock-lists/pull/1248

**Review by antonok-edm — APPROVED** [review]

**Pull request merged**